java/src/org/openqa/selenium/grid/data/NodeStatus.java

```diff
@@ -204,6 +204,65 @@ public long getLastSessionCreated() {
         .orElse(0);
   }
 
+  public String getBrowserVersion() {
+    return slots.parallelStream()
+        .map(slot -> slot.getStereotype().getBrowserVersion())
+        .filter(Objects::nonNull)
+        .min(NodeStatus::semVerComparator)
+        .orElse("");
+  }
+
+  public static int semVerComparator(String v1, String v2) {
+    // Custom semver comparator with empty strings first
+    if (v1.isEmpty() && v2.isEmpty()) return 0;
+    if (v1.isEmpty()) return -1; // Empty string comes first
+    if (v2.isEmpty()) return 1;
+
+    String[] parts1 = v1.split("\\.");
+    String[] parts2 = v2.split("\\.");
+
+    int maxLength = Math.max(parts1.length, parts2.length);
+    for (int i = 0; i < maxLength; i++) {
+      String part1 = i < parts1.length ? parts1[i] : "0";
+      String part2 = i < parts2.length ? parts2[i] : part1;
```

Comment:
This code shouldn't be here. It probably belongs to the `DefaultSlotMatcher`.

Reply:
I moved it to `CapabilitiesComparator` for reusable. Since this is used for both sorting in selector and slot matcher